pkg/virt-controller/watch/vm_test.go

```diff
@@ -308,6 +314,39 @@ var _ = Describe("VirtualMachine", func() {
 			mockQueue.Wait()
 		}
 
+		It("should update conditions when failed creating DataVolume for virtualMachineInstance", func() {
+			vm, _ := DefaultVirtualMachine(true)
+			vm.Spec.Template.Spec.Volumes = append(vm.Spec.Template.Spec.Volumes, virtv1.Volume{
+				Name: "test1",
+				VolumeSource: virtv1.VolumeSource{
+					DataVolume: &virtv1.DataVolumeSource{
+						Name: "dv1",
+					},
+				},
+			})
+			vm.Spec.DataVolumeTemplates = append(vm.Spec.DataVolumeTemplates, virtv1.DataVolumeTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "dv1",
+				},
+			})
+			vm.Status.PrintableStatus = virtv1.VirtualMachineStatusStopped
+			addVirtualMachine(vm)
+
+			shouldFailDataVolumeCreationNoResourceFound()
+
+			vmInterface.EXPECT().UpdateStatus(context.Background(), gomock.Any()).Times(1).Do(func(ctx context.Context, obj interface{}) {
```

Comment:
Could we use a fake client?

Reply:
what do you mean?

Reply:
This is mock, I was suggesting using fake as we do for cdi client. But that might be more work.

Reply:
I did the same as done in other tests here. works good :)